src/mechanics/interpolation/InterpolationBrickLobatto.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include "mechanics/interpolation/InterpolationSimple.h"
+#include "mechanics/elements/SpectralShapeFunctions.h"
+
+namespace NuTo
+{
+class InterpolationBrickLobatto : public InterpolationSimple
+{
+public:
+    InterpolationBrickLobatto(int dofDimension, int order)
+        : mDofDimension(dofDimension)
+    {
+        mNodes = ShapeFunctions1D::NodeCoordinatesTrussLobatto(order);
+    }
+
+    std::unique_ptr<InterpolationSimple> Clone() const override
+    {
+        return std::make_unique<InterpolationBrickLobatto>(*this);
+    }
+
+    ShapeFunctions GetShapeFunctions(const NaturalCoords& naturalIpCoords) const override
+    {
+        Eigen::VectorXd result(GetNumNodes());
+        const std::vector<double> shapes = ShapeFunctions3D::ShapeFunctionsBrickLagrange(naturalIpCoords, mNodes);
```

Comment:
I would prefer defining your `ShapeFunctions3D` to return `Eigen` types directly.